codec/src/main/java/io/netty/handler/codec/compression/JdkZlibEncoder.java

```diff
@@ -200,42 +201,50 @@ protected void encode(ChannelHandlerContext ctx, ByteBuf uncompressed, ByteBuf o
 
         int offset;
         byte[] inAry;
-        if (uncompressed.hasArray()) {
-            // if it is backed by an array we not need to to do a copy at all
-            inAry = uncompressed.array();
-            offset = uncompressed.arrayOffset() + uncompressed.readerIndex();
-            // skip all bytes as we will consume all of them
-            uncompressed.skipBytes(len);
-        } else {
-            inAry = new byte[len];
-            uncompressed.readBytes(inAry);
-            offset = 0;
-        }
+        ByteBuf heapBuf = null;
```

Comment:
make this final and assign in the `if / else ` blocks

Reply:
@normanmaurer ,
> make this final and assign in the if / else blocks

I introduced a `try / finally` so that we would not leak the reference to the allocated heap buffer; this accounts for the formatting changes above, and it also confuses the Java compiler when making `heapBuf` final; see attached screenshot.
![Screenshot from 2021-12-06 10-00-16](https://user-images.githubusercontent.com/13126037/144897921-5aff2f72-9d7c-40a9-9012-072bf18b20e4.png)

Reply:
got it...